File: Lab-Question-Answer/Lab12.java
import java.util.Scanner;


class Person {
   protected String name;
   protected int age;


   public Person(String name, int age) {
       this.name = name;
       this.age = age;
   }


   public void display() {
       System.out.println("Name: " + name);
       System.out.println("Age: " + age);
   }
}


class Employee extends Person {
   private double salary;


   public Employee(String name, int age, double salary) {
       super(name, age);
       this.salary = salary;
   }


   public void displaySalary() {
       System.out.println("Salary: $" + salary);
   }
}


public class Lab12 {
   public static void main(String[] args) {
       Scanner scanner = new Scanner(System.in);


       System.out.print("What is your name: ");
       String name = scanner.next();
       System.out.print("Your age: ");
       int age = scanner.nextInt();
       System.out.print("Your salary: ");
       double salary = scanner.nextDouble();


       System.out.println("---------------");


       Employee emp = new Employee(name, age, salary);
       emp.display();
       emp.displaySalary();


       scanner.close();
   }
}

File: Lab-Question-Answer/Lab12.cpp
#include <iostream>
#include <string>
using namespace std;
class Person {
protected:
   string name;
   int age;
public:
   Person(string name, int age) : name(name), age(age) {}
   void display() {
       cout << "Name: " << name << endl;
       cout << "Age: " << age << endl;
   }
};
class Employee : public Person {
private:
   double salary;
public:
   Employee(string name, int age, double salary) : Person(name, age), salary(salary) {}
   void displaySalary() {
       cout << "Salary: $" << salary << endl;
   }
};


int main() {
   string name;
   int age;
   double salary;


   cout<<"What is your name :";
   cin>>name;
   cout<<"Your age : ";
   cin>>age;
   cout<<"Your salary : ";
   cin>>salary;


   cout<<"---------------"<<endl;
   Employee emp(name, age, salary);
   emp.display();
   emp.displaySalary();
   return 0;
}
